### ISSUE TITLE
Allow MAX function as a CONCAT argument (improvement?)

### ISSUE THREAD

**Issue by stphane** [doctrine/doctrine2#7205]
I recently migrated our project from doctrine _v2.5.6_ to _v2.6.1_. and ran into an error saying:
> Expected StateFieldPathExpression | string | InputParameter | FunctionsReturningStrings | AggregateExpression, got 'MAX'

… with the query being:

```
$QB = $this->_em->createQueryBuilder();
$QB ->addSelect("CONCAT(t2.dagId, MAX(t1.executionDate))")
    ->from(TaskInstance::class, 't1')
    ->join('t1.dag', 't2')
    ->addGroupBy('t2.dagId')
    ;


// Calling either getResults() or getSQL raises the error.
print_r($QB->getQuery()->getSQL());
```
After looking at `Doctrine\ORM\Query\Parser::StringPrimary()`, I ended up nesting `MAX(…)` call within a `IF` to fix the error:
```
$QB ->addSelect("CONCAT(t2.dagId, IF(1=1, MAX(t1.executionDate)))")
```
Is such a call rejection normal/intended or should this problem be addressed ?

**Comment by stof** [comment]
shouldn't `MAX` be an `AggregateExpression` ?

**Comment by stphane** [comment]
That's what I thought since it is an aggregate function. It feels surprising that such functions are not considered to be valid tokens in this context.

**Comment by NothingWeAre** [comment]
After encountering similar problem, and looking at code i've noticed that StringPrimary no longer processes aggregate functions, this change happened somwhere betwean 2.5.14 - 2.6.1
Old function looked like:
``` 
            case Lexer::T_CASE:
            case Lexer::T_COALESCE:
            case Lexer::T_NULLIF:
                return $this->CaseExpression();

            default:
                if ($this->isAggregateFunction($lookaheadType)) {
                    return $this->AggregateExpression();
                }
        }
        $this->syntaxError(
            'StateFieldPathExpression | string | InputParameter | FunctionsReturningStrings | AggregateExpression'
        );
```
while new one missing part responsible for accepting aggregate expressions:
```
            case Lexer::T_CASE:
            case Lexer::T_COALESCE:
            case Lexer::T_NULLIF:
                return $this->CaseExpression();
        }
        $this->syntaxError(
            'StateFieldPathExpression | string | InputParameter | FunctionsReturningStrings | AggregateExpression'
        );
```

I did not find any mentions of this in documentation (but I could have missed it), so i think this is regression.

**Comment by Majkl578** [comment]
This has been fixed in [2.6.2](https://github.com/doctrine/doctrine2/releases/tag/v2.6.2) via https://github.com/doctrine/doctrine2/pull/7296.
